### PR TITLE
Add FPS indicator

### PR DESCRIPTION
**Pull request by 0xsuu**

### CODE REVIEW

File: include/ABEObject.h [hunk 1]
```diff
@@ -38,16 +39,20 @@ class ABEObject {
     void setPosition(float x, float y);
     void pushPosition(float x, float y);
     std::pair<float, float> popPosition();
+    float getX() { return mPositionStack.back().first; }
+    float getY() { return mPositionStack.back().second; }
```
Comment (0xsuu) [inline include/ABEObject.h:43]: sb

File: src/ABETexture.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        offset += size;
+        width  /= 2;
+        height /= 2;
+
+        if(width < 1) width = 1;
+        if(height < 1) height = 1;
+    }
+
+    free(buffer);
+}
```
Comment (0xsuu) [inline src/ABETexture.cpp]: @ZongyuYin Gai

File: src/ABETexture.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    for (unsigned int level = 0; level < mipMapCount && (width || height); ++level) {
+        unsigned int size = ((width + 3) / 4) * ((height + 3) / 4) * blockSize;
+        glCompressedTexImage2D(GL_TEXTURE_2D, level, format, width, height,
+                               0, size, buffer + offset);
+
+        offset += size;
+        width  /= 2;
+        height /= 2;
+
+        if(width < 1) width = 1;
```
Comment (0xsuu) [inline src/ABETexture.cpp]: @ZongyuYin Gai

File: src/ABETexture.cpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        if(height < 1) height = 1;
```
Comment (0xsuu) [inline src/ABETexture.cpp]: @ZongyuYin Gai

**Comment by 0xsuu** [comment]
@ZongyuYin Try a new font bitmap instead the current one(Holstein.DDS). And rename it to DefaultFont.dds.

**Review by Nniy — APPROVED** [review]
Perfect

**Pull request merged**